homeassistant/components/daikin/__init__.py

```diff
@@ -90,6 +90,7 @@
     try:
         with async_timeout.timeout(10):
             device = await hass.async_add_executor_job(Appliance, host)
```

Comment:
Do we still need to run this in the executor thread pool? How much of the library is async now?

Reply:
The `__init__()` look up devices if just a common name is given (afaik can't `__init__` be async).

Reply:
`__init__` method can't be a coroutine, but if it doesn't do non asyncio I/O we can run it in the event loop directly.

But it looks like it still does some network look ups, ie I/O.

It's generally bad form to have side effects in init methods. I would recommend moving those out of the init method of `Appliance`. We will still have to use the executor though for those, so it doesn't relieve us of that.

Reply:
Look into UDP transport using asyncio to replace the non asyncio discovery of pydaikin:
https://docs.python.org/3/library/asyncio-protocol.html

Reply:
I don't like to break the current Appliance too much, but if I make `session` optional I could skip the discovery part that is not used by home assistant if `session` is set, what do you think about that?

Reply:
> Ah, since we provide `host` to the `Appliance`, the discovery isn't done at all. This is regardless of sesssion is optional or not, right?

Yes more or less, if host is an ip it currently do no discovery. But if it is a DNS-name it will do discovery and then DNS-lookup. We could skip that if session is set as i suppose that aiohttp can handle DNS-names and assume that home assistant only will send id as ip/dns-name.

Reply:
I think you know this best.

For the long term I would recommend making the library fully async and clarify the api. Having to instantiate the appliance in the thread pool while the other appliance methods are coroutines is unexpected.